uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -162,31 +153,117 @@ define([
     }());
 
     /**
-     * Calculate change in balance, it updates local balance variables.
-     *
-     * @return {Promise} empty promise if all required params are not set
-     *   otherwise promise from server
+     * Adjusts dates in UI.
+     * "To" date must not be earlier than "From" date in case of a Single day request.
+     * "To" date must be after "From" date in case of a Multiple day request.
      */
-    function calculateBalanceChange () {
-      vm._setDateAndTypes();
-      vm._toggleBalance();
+    function adjustDatesInUI () {
+      var toDate = moment(vm.uiOptions.toDate);
 
-      if (!vm._canCalculateChange()) { return $q.resolve(); }
+      if (vm.uiOptions.toDate) {
+        if (!vm.uiOptions.multipleDays) {
```

Comment:
@AkA84 I am not really sure what do you mean by 
> The first half shouldn't be required at all

Reply:
> The first half shouldn't be required at all

because 

> should this happen automatically the moment user switches modes?

If the moment the user switched mode the "to" date gets assigned the value of "from" automatically, then there is no need to check if the two are different here

Reply:
Let's talk about this in a call.

Reply:
This is refactored.